data/mods/ssb/abilities.ts

```diff
@@ -196,887 +472,1759 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
+			const moveData = pokemon.getMoveData(move.id);
+			if (!moveData) return;
+			// Lost 1 PP due to move usage, restore 0.5 PP to make it so that only 0.5 PP
+			// would be used.
+			moveData.pp = (Math.round(moveData.pp * 100) + 50) / 100;
+		},
+		isNonstandard: "Custom",
+		gen: 8,
 	},
-	// SamJo
-	thiccerfat: {
-		desc: "If a Pokemon uses a Fire- or Ice-type attack against this Pokemon, that Pokemon's attacking stat is halved when calculating the damage to this Pokemon. This Pokemon cannot be burned. Gaining this Ability while burned cures it.",
-		shortDesc: "Fire/Ice damage against this Pokemon has a halved attacking stat. Burn immunity.",
-		name: "Thiccer Fat",
+
+	// Segmr
+	skilldrain: {
+		desc: "While this Pokemon is active, no moves will trigger their secondary effects, and moves that cause the user to switch out will no longer do so.",
+		shortDesc: "While active: no secondary effects, moves can't switch out.",
+		name: "Skill Drain",
+		onAnyModifyMove(move) {
+			delete move.secondaries;
+		},
+		// afterSecondarySelf and switch nullifying handled in ssb/scripts.ts
 		isNonstandard: "Custom",
-		onSourceModifyAtkPriority: 6,
-		onSourceModifyAtk(atk, attacker, defender, move) {
-			if (move.type === 'Ice' || move.type === 'Fire') {
-				this.debug('Thiccer Fat weaken');
-				return this.chainModify(0.5);
+		gen: 8,
+	},
+
+	// sejesensei
+	trashconsumer: {
+		desc: "This Pokemon is immune to Poison-type moves and restores 1/4 of its maximum HP, rounded down, when hit by a Poison-type move. Pokemon making contact with this Pokemon lose 1/8 of their maximum HP, rounded down.",
+		shortDesc: "Poison Absorb + Rough Skin",
+		name: "Trash Consumer",
+		onTryHit(target, source, move) {
+			if (target !== source && move.type === 'Poison') {
+				if (!this.heal(target.baseMaxhp / 4)) {
+					this.add('-immune', target, '[from] ability: Trash Consumer');
+				}
+				return null;
 			}
 		},
-		onSourceModifySpAPriority: 5,
-		onSourceModifySpA(atk, attacker, defender, move) {
-			if (move.type === 'Ice' || move.type === 'Fire') {
-				this.debug('Thiccer Fat weaken');
-				return this.chainModify(0.5);
+		onDamagingHitOrder: 1,
+		onDamagingHit(damage, target, source, move) {
+			if (move.flags['contact']) {
+				this.damage(source.baseMaxhp / 8, source, target);
 			}
 		},
-		onUpdate(pokemon) {
-			if (pokemon.status === 'brn') {
-				this.add('-activate', pokemon, 'ability: Thiccer Fat');
-				pokemon.cureStatus();
+		isNonstandard: "Custom",
+		gen: 8,
+	},
+
+	// Shadecession
+	shadydeal: {
+		desc: "On switch-in, this Pokemon boosts a random stat other than Special Attack by 1 stage and gains 2 random type immunities that are displayed to the opponent.",
+		shortDesc: "On switch-in, gains random +1 to non-SpA, 2 random immunities.",
+		onStart(pokemon) {
+			const stats: BoostName[] = [];
+			let stat: BoostName;
+			for (stat in pokemon.boosts) {
+				const noBoost: string[] = ['accuracy', 'evasion', 'spa'];
```

Comment:
```suggestion
			for (const stat in pokemon.boosts) {
				const noBoost = ['accuracy', 'evasion', 'spa'];
```
Duck typing is fine here.

Reply:
Even without the `for` loop thing, it should recognize that as a string[]